nautobot/dcim/filters.py

```diff
@@ -159,21 +170,85 @@ class SiteFilterSet(NautobotFilterSet, TenancyFilterSet, StatusModelFilterSetMix
         to_field_name="slug",
         label="Region (slug)",
     )
+    has_circuit_terminations = RelatedMembershipBooleanFilter(
+        field_name="circuit_terminations",
+        label="Has circuit terminations",
+    )
+    has_devices = RelatedMembershipBooleanFilter(
+        field_name="devices",
+        label="Has devices",
+    )
+    powerpanels = django_filters.ModelMultipleChoiceFilter(
+        field_name="powerpanel",
+        queryset=PowerPanel.objects.all(),
+        label="Power panels",
+    )
+    has_powerpanels = RelatedMembershipBooleanFilter(
+        field_name="powerpanel",
+        label="Has power panels",
+    )
+    rack_groups = NaturalKeyMultipleChoiceFilter(
+        queryset=RackGroup.objects.all(),
+        label="Rack groups (slug or ID)",
+    )
+    has_rack_groups = RelatedMembershipBooleanFilter(
+        field_name="rack_groups",
+        label="Has rack groups",
+    )
+    has_racks = RelatedMembershipBooleanFilter(
+        field_name="racks",
+        label="Has racks",
+    )
+    has_prefixes = RelatedMembershipBooleanFilter(
+        field_name="prefixes",
+        label="Has prefixes",
+    )
+    vlan_groups = NaturalKeyMultipleChoiceFilter(
+        queryset=VLANGroup.objects.all(),
+        label="Vlan groups",
+    )
+    has_vlan_groups = RelatedMembershipBooleanFilter(
+        field_name="vlan_groups",
+        label="Has vlan groups",
+    )
+    has_vlans = RelatedMembershipBooleanFilter(
+        field_name="vlans",
+        label="Has vlans",
+    )
+    has_clusters = RelatedMembershipBooleanFilter(
+        field_name="clusters",
+        label="Has clusters",
+    )
+    time_zone = django_filters.MultipleChoiceFilter(
+        choices=[(str(obj), name) for obj, name in TimeZoneField().choices],
+        label="Time zone",
+        null_value="",
+    )
     tag = TagFilter()
 
     class Meta:
         model = Site
         fields = [
```

Comment:
What keeps us from setting fields to `"__all__"` here at this point?

Reply:
We might be able to now. I'll test

Reply:
It looks like it's not picking up any of the reverse relationships with `"__all__"`:
- `devices`
- `prefixes`
- `racks`
- `vlans`

When you set `"__all__"` [django-filter](https://github.com/carltongibson/django-filter/blob/48cad7b31139a56bf7c864484bb63e132f2a30ab/django_filters/utils.py#L123) is only looking at the `model._meta.fields` and `model._meta.many_to_many` properties which are both [forward properties](https://github.com/django/django/blob/08e6073f878264a0c091da0d3db456820252ef6c/django/db/models/options.py#L65). We could write our own method to resolve the forward and reverse fields to populate the filterset `meta.fields` or open an issue with django-filter to see if they will add reverse fields to `"__all__"`.